### PR TITLE
EVG-6311 check project has commit-queue enabled

### PR DESCRIPTION
**Pull request by ablack12**
Check project has commit-queue enabled before creating a patch.

### CODE REVIEW

File: operations/commit_queue.go [hunk 1]
```diff
@@ -319,6 +319,9 @@ func (p *mergeParams) uploadMergePatch(conf *ClientSettings, ac *legacyClient) e
 		}
 		return errors.Wrap(err, "can't get project ref")
 	}
+	if !ref.CommitQueue.Enabled {
```
Comment (ybrill) [inline operations/commit_queue.go]: maybe we should also check the project is using the CLI queue?

Comment (ablack12) [reply operations/commit_queue.go]: oh very good point

File: operations/commit_queue.go [hunk 2]
```diff
@@ -319,6 +319,9 @@ func (p *mergeParams) uploadMergePatch(conf *ClientSettings, ac *legacyClient) e
 		}
 		return errors.Wrap(err, "can't get project ref")
 	}
+	if !ref.CommitQueue.Enabled && ref.CommitQueue.PatchType == commitqueue.CLIPatchType {
```
Comment (ybrill) [inline operations/commit_queue.go]: I think we would want to check `!(ref.CommitQueue.Enabled && ref.CommitQueue.PatchType == commitqueue.CLIPatchType)` or the equivalent.

Comment (ablack12) [reply operations/commit_queue.go]: Sorry, apparently I don't pay close enough attention when I juggle tickets :)

**Pull request merged**